src/main/java/de/dennisguse/opentracks/services/handlers/TrackPointCreator.java

```diff
@@ -114,16 +115,28 @@ public void stop() {
         this.context = null;
     }
 
+    public void onChange(@NonNull Location location) {
+        synchronized (this) {
+            TrackPoint trackPoint = new TrackPoint(location, createNow());
+            addSensorData(trackPoint);
+            onNewTrackPoint(trackPoint);
+        }
+    }
+
     /**
      * Got a new TrackPoint from Bluetooth only; contains no GPS location.
      */
     @Override
-    public void onChange(SensorDataSet sensorDataSet) {
-        onNewTrackPoint(new TrackPoint(TrackPoint.Type.SENSORPOINT, createNow()));
+    public void onChange(@NonNull SensorDataSet unused) {
+        synchronized (this) {
```

Comment:
A synchronized block for the whole method? Can this be simplified by adding the synchronized keyword to the method? I think it' equivalent.

Reply:
Good question; not sure.
Just checked the documentation and making the onChange()-methods would have the same effect.
Was not aware of this - thanks for the hint :D

https://docs.oracle.com/javase/tutorial/essential/concurrency/syncmeth.html

Reply:
Just added a commit changing this :D